rskj-core/src/test/java/co/rsk/peg/BridgeSupportSigHashTest.java

```diff
@@ -0,0 +1,292 @@
+import co.rsk.bitcoinj.core.Context;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.TransactionOutput;
+import co.rsk.bitcoinj.core.UTXO;
+import co.rsk.bitcoinj.wallet.Wallet;
+import co.rsk.blockchain.utils.BlockGenerator;
+import co.rsk.config.BridgeConstants;
+import co.rsk.config.BridgeMainNetConstants;
+import co.rsk.peg.bitcoin.BitcoinUtils;
+import co.rsk.test.builders.BridgeSupportBuilder;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ActivationConfigsForTest;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+import org.ethereum.core.Transaction;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class BridgeSupportSigHashTest {
+
+    private static final BridgeConstants bridgeMainnetConstants = BridgeMainNetConstants.getInstance();
+    private static final NetworkParameters btcMainnetParams = bridgeMainnetConstants.getBtcParams();
+
+    private static Stream<Arguments> pegoutTxIndexArgsProvider() {
+        return Stream.of(
+            Arguments.of(ActivationConfigsForTest.fingerroot500().forBlock(0)),
+            Arguments.of(ActivationConfigsForTest.tbd600().forBlock(0))
+        );
+    }
+    @ParameterizedTest
+    @MethodSource("pegoutTxIndexArgsProvider")
+    void test_pegoutTxIndex_when_pegout_batch_is_created(ActivationConfig.ForBlock activations) throws IOException {
+        // Arrange
+        BridgeStorageProvider provider = mock(BridgeStorageProvider.class);
+
+        List<UTXO> fedUTXOs = PegTestUtils.createUTXOs(10, bridgeMainnetConstants.getGenesisFederation().getAddress());
+        when(provider.getNewFederationBtcUTXOs())
+            .thenReturn(fedUTXOs);
+
+        when(provider.getReleaseRequestQueue())
+            .thenReturn(new ReleaseRequestQueue(PegTestUtils.createReleaseRequestQueueEntries(3)));
+
+        PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = new PegoutsWaitingForConfirmations(new HashSet<>());
+        when(provider.getPegoutsWaitingForConfirmations())
+            .thenReturn(pegoutsWaitingForConfirmations);
+
+        when(provider.getPegoutsWaitingForSignatures())
+            .thenReturn(new TreeMap<>());
```

Comment:
This Is Used In a Couple Of Tests And Can Be Moved Into @ before Method

Reply:
Done